modules/web-console/web-console-notification/src/test/java/org/apache/ignite/console/notification/services/MailServiceTest.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.ignite.console.notification.model.Recipient;
+import org.apache.ignite.console.notification.config.MessagesProperties;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.MessageSource;
+import org.springframework.mail.javamail.JavaMailSender;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Mail service test.
+ */
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest
+public class MailServiceTest {
+    /** Message source. */
+    @Autowired
+    private MessageSource msgSrc;
+
+    /** JavaMail sender. */
+    @Mock
+    private JavaMailSender mailSnd;
+
+    /** Argument capture  */
+    @Captor
+    private ArgumentCaptor<MimeMessage> captor;
+
+    /** Mail service. */
+    private MailService srvc;
+
+    /** */
+    @Before
+    public void setup() {
+        MessagesProperties cfg = new MessagesProperties();
+
+        srvc = new MailService(msgSrc, mailSnd, cfg);
+
+        when(mailSnd.createMimeMessage())
+            .thenReturn(new MimeMessage(Session.getDefaultInstance(new Properties())));
+    }
+
+    /** Test send e-mail. */
+    @Test
+    public void shouldSendEmail() throws MessagingException, IOException {
+        Notification notification = new Notification(
+            "http://test.com",
+            new TestRecipient(),
+            null
+        );
+
+//        notification.setSubject("subject");
+//        notification.setMessage("text");
+
+        srvc.send(notification);
+
+        verify(mailSnd).send(captor.capture());
+
+        MimeMessage msg = captor.getValue();
+        
+        assertEquals("subject", msg.getSubject());
+        assertEquals("text", msg.getContent());
+    }
+
+    /** */
+    @Test
+    public void shouldSendEmailWithExpressionInSubject() throws MessagingException, IOException {
+        Notification notification = new Notification(
+            "http://test.com",
+            new TestRecipient(),
+            null
+        );
+
+//        notification.setSubject("Hello ${recipient.firstName} ${recipient.lastName}! subject");
+//        notification.setMessage("text");
+
+        srvc.send(notification);
+
+        verify(mailSnd).send(captor.capture());
+
+        MimeMessage msg = captor.getValue();
+
+        assertEquals("Hello firstName lastName! subject", msg.getSubject());
+        assertEquals("text", msg.getContent());
+    }
+
+    /** */
+    private static class TestRecipient implements Recipient {
+        /** First name. */
+        private String fn = "firstName";
+
+        /** Last name. */
+        public String lastName = "lastName";
```

Comment:
This variable is not used, lets remove?

Reply:
This code is valid, it checks access to methods or properties in message templates.